pkg/server/grpc/grpc.go

```diff
@@ -67,17 +73,40 @@ func New(logger log.Logger, reg prometheus.Registerer, tracer opentracing.Tracer
 		return status.Errorf(codes.Internal, "%s", p)
 	}
 
+	loggingOpts := []grpc_logging.Option{
+		grpc_logging.WithDecider(func(_ string) grpc_logging.Decision {
+			return LogDecision[requestLoggingDecision]
+		}),
+	}
+
+	tagsOption := []tags.Option{
+		tags.WithFieldExtractor(func(_ string, _ interface{}) map[string]string {
+
+			entropy := ulid.Monotonic(rand.New(rand.NewSource(time.Now().UnixNano())), 0)
+			reqID := ulid.MustNew(ulid.Timestamp(time.Now()), entropy)
+
+			tagMap := make(map[string]string)
+			tagMap["request-id"] = reqID.String()
```

Comment:
What if request ID is already there passed? :thinking:

Reply:
Yeah, good catch! I didn't handle the case, doing right away!